src/main/bootstrap/cache_validator.rs

```diff
@@ -149,6 +125,15 @@ impl State<CrustData> for CacheValidator {
     }
 }
 
+/// Schedules tests for inactive cached bootstrap peers.
+pub fn test_inactive_cached_peers(core: &mut EventLoopCore, poll: &Poll, peers: HashSet<PeerInfo>) {
+    if let Some(state) = core.get_state(EventToken::BootstrapCacheValidator.into()) {
+        if let Some(validator) = state.borrow_mut().as_any().downcast_mut::<CacheValidator>() {
```

Comment:
It is to avoid this downcast i was [mentioning](https://github.com/maidsafe/crust/pull/1120#discussion_r256052331) about having it in the Core's userdata directly. Would that be a better pattern ? Because as we can see there's not really anything from the `State` trait that this is using so why even make it a `State` was my reasoning.

Reply:
I actually experimented with this in the morning and the problem was that I had to terminate the `GetExtAddr` request states. I thought of doing that during `drop()` of `CacheValidator` but I can't access `poll` and `core`, hence this implementation.

Any ideas how I could nicely terminate the [sent_requests](https://github.com/maidsafe/crust/pull/1120/files#diff-7e42e856fb45c217a3ce4b3d9f24650bR33) without `CacheValidator` being a `State`?

Reply:
Can it not have a non-state terminate member function just in the `impl CacheValidator` ?

Reply:
It can, but then there's no one preventing us from leaking those states except explicitly documenting that such destructor MUST be called.